### PR TITLE
ILL SANS Absolute Scale

### LINKED ISSUE #27918: ILL SANS Absolute scale
It turns out most often the water run is measured with a different distance/collimation than the sample.
So when normalising by water, one has to rescale to account for different fluxes in order to get the absolute normalisation right.

### PR DESCRIPTION
**Pull request by gvardany**
**Description of work.**
This PR makes it possible to do correct absolute normalisation even if the water and the sample runs are measured at different distances, and yet even if there is no flux measurement at the water distance.

**To test:**
Code review and replay of system test should suffice `./systemtest -R SANSILL`

Fixes #27918 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by Harrietbrown — CHANGES_REQUESTED** [review]
I am getting a failing system test `SANSILLAutoProcessTest.D33_AutoProcess_Test`

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLReduction.py [hunk 1]
```diff
@@ -357,15 +348,61 @@ def _process_sample(self, ws):
...
+            if log_val != log_val_ref:
+                raise RuntimeError(message)
+            elif log_val == 'False':
+                self._do_rescale_flux(ws, ref_ws)
+        else:
+            self._do_rescale_flux(ws, ref_ws)
+
+    def _do_rescale_flux(self, ws, ref_ws):
+        """
+            Scales ws by the magic flux factor wrt the reference
```
[the remark this answers is not in the document]
Comment (gvardany) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLReduction.py]: There is no much magic at the end, I'll remove it :)

**Comment by gvardany** [comment]
What is the failure exactly?

**Comment by Harrietbrown** [comment]
```
SANSILLIntegration-[Error] Error in execution of algorithm SANSILLIntegration:
SANSILLIntegration-[Error] Index out of range
SANSILLIntegration-[Error]   at line 179 in 'C:/Users/wey38795/Documents/mantid_builds/Master/raw/mantid/Framework/PythonInterface/plugins/algorithms\WorkflowAlgorithms\SANSILLIntegration.py'
SANSILLAutoProcess-[Error] Error in execution of algorithm SANSILLAutoProcess:
SANSILLAutoProcess-[Error] Index out of range
SANSILLAutoProcess-[Error]   at line 179 in 'C:/Users/wey38795/Documents/mantid_builds/Master/raw/mantid/Framework/PythonInterface/plugins/algorithms\WorkflowAlgorithms\SANSILLIntegration.py'
SANSILLAutoProcess-[Error]   at line 299 in 'C:/Users/wey38795/Documents/mantid_builds/Master/raw/mantid/Framework/PythonInterface/plugins/algorithms\WorkflowAlgorithms\SANSILLAutoProcess.py'
SANSILLAutoProcess-[Error]   caused by line 461 in 'C:/Users/wey38795/Documents/mantid_builds/Master/raw/mantid/Framework/PythonInterface/plugins/algorithms\WorkflowAlgorithms\SANSILLAutoProcess.py'
SANSILLAutoProcess-[Error]   caused by line 1171 in 'c:\users\wey38795\documents\mantid_builds\master\raw\mantid\framework\pythoninterface\mantid\simpleapi.py'
Traceback (most recent call last):
  File "C:\Users\wey38795\AppData\Local\Temp\tmpi2_zo3ge", line 24, in <module>
    systest.execute()
  File "C:\Users\wey38795\Documents\mantid_builds\Master\raw\mantid\Testing\SystemTests\lib\systemtests\systemtesting.py", line 225, in execute
    self.runTest()
  File "C:/Users/wey38795/Documents/mantid_builds/Master/raw/mantid/Testing/SystemTests/tests/analysis\SANSILLAutoProcessTest.py", line 101, in runTest
    SANSILLAutoProcess(
  File "c:\users\wey38795\documents\mantid_builds\master\raw\mantid\framework\pythoninterface\mantid\simpleapi.py", line 1171, in algorithm_wrapper
    algm.execute()
RuntimeError: Index out of range
  at line 179 in 'C:/Users/wey38795/Documents/mantid_builds/Master/raw/mantid/Framework/PythonInterface/plugins/algorithms\WorkflowAlgorithms\SANSILLIntegration.py'
  at line 299 in 'C:/Users/wey38795/Documents/mantid_builds/Master/raw/mantid/Framework/PythonInterface/plugins/algorithms\WorkflowAlgorithms\SANSILLAutoProcess.py'
  caused by line 461 in 'C:/Users/wey38795/Documents/mantid_builds/Master/raw/mantid/Framework/PythonInterface/plugins/algorithms\WorkflowAlgorithms\SANSILLAutoProcess.py'
  caused by line 1171 in 'c:\users\wey38795\documents\mantid_builds\master\raw\mantid\framework\pythoninterface\mantid\simpleapi.py'
```

**Comment by gvardany** [comment]
Does it go away if you wipe the IDF cache? It's `~/.mantid/instrument` on Unix, on Windows I think it's somewhere in APP DATA.

**Comment by Harrietbrown** [comment]
OK that has gotten the tests passing.

**Review by Harrietbrown — APPROVED** [review]
All system tests are now passing on jenkins and localy and the code passes review.

Good job.

**Pull request merged**